dex/farm_with_lock/README.md

```diff
@@ -1,28 +1,25 @@
-# Farm Smart Contract
+# Farm With Lock Smart Contract
 
-This document presents how one can deploy and configure a Farm contract.
-The bigger picture about what a Farm contract can do can be found in the Repository's Root Readme.
+## Abstract
 
-## Deployment
-
-The Farm Contract can be deployed using `erdpy` and using the interraction snippets.
+This contract is very similar with the regular Farm contract. It is recommended that you go though its README first, as this doc will cover only the differences.
 
-The init parameters are:
+## Introduction
 
-- router_address. The address of the router. This may seem a bit useless because the router cannot configure a Farm contract, but we'll keep the address there in case it will be needed.
+This contract will rewards its participants with Locked MEX instead of MEX. Locked MEX is a META-ESDT with certain metadata stored as attributes. Because the META-ESDT create role can be held by only one address per shard, instead of creating the Locked MEX itself, the contract will do a request to Locked MEX Factory to create and forward the tokens to the user, on its behalf.
```

Comment:
```suggestion
This contract will reward its participants with Locked MEX instead of MEX. Locked MEX is a META-ESDT with certain metadata stored as attributes. Because the META-ESDT create role can be held by only one address per shard, instead of creating the Locked MEX itself, the contract will do a request to Locked MEX Factory to create and forward the tokens to the user, on its behalf.
```